### PR TITLE
allow setHash run if proceed==true

### PR DESCRIPTION
**Pull request by fremorie**

### CODE REVIEW

File: src/babel.js [hunk 1]
```diff
@@ -140,13 +140,21 @@ module.exports = () => {
...
   return ({
     pre(state) {
       componentsList = new Set()
       pathsList = new Set()
       hashmap = {}
       /** don't apply reselector for files that were already proceed */
       proceed = (state.ast.comments || []).some(x => x.value.includes('__reselector__start__::'))
+
+      if (proceed) {
+        dataFromComment = JSON.parse(
```
Comment (lttb) [inline src/babel.js:154]: I think you can try to use this helper:

https://github.com/lttb/reselector/blob/29e4c7c509b45fe9eb4d124b67de7757fbf4c168/src/utils.js#L139

File: src/babel.js [hunk 2]
```diff
@@ -155,7 +163,25 @@ module.exports = () => {
...
+              loc: dataFromComment[names[i]].loc,
+            })
+          }
+
+          return
+        }
+
+        if (proceed) {
+          return
+        }
```
Comment (lttb) [inline src/babel.js:184]: Would mind to make it a bit simpler maybe?

```suggestion
        const { filename } = file.opts

        if (proceed) {
          if (opts.setHash) {
            Object.entries(dataFromComment).forEach(([name, data])=> {
              opts.setHash({
                filename,
                name,
                id: data.id,
                loc: data.loc,
              })
            })
          }
        
          return;
        }
```

File: src/resolve.js [hunk 3]
```diff
@@ -58,7 +58,7 @@ const createResolve = (config) => {
       const hashmap = getHashmapFromComment(content)
 
       if (hashmap) {
-        cache[filename] = hashmap
+        cache[filename] = { [hashmap.name]: { [config.name]: hashmap.id } }
```
Comment (lttb) [inline src/resolve.js:61]: `cache[filename]` is kind of `{ComponentA: {'data-id': 'hash'}, ComponentB: {'data-id': 'hash'}, ...}`
so you need to iterate through the `hashmap` fields and map values to something like `{config.name}: data.id`

File: src/resolve.js [hunk 4]
```diff
@@ -52,13 +52,19 @@ const createResolve = (config) => {
 
   const cache = {}
 
-  const resolve = (filename) => {
+  const resolve = (filename, contentFromFile) => {
     if (!cache[filename]) {
-      const content = readFileSync(filename).toString()
+      // contentFromFile is here for testing purposes only
+      const content = contentFromFile || readFileSync(filename).toString()
```
Comment (fremorie) [inline src/resolve.js:58]: every mock I've tried for `fs.readFileSync` inevitably broke files from functions from the `resolve` module.
this is the only solution that worked.

**Pull request merged**